### PR TITLE
MRG: Replace tempita with Jinja2

### LINKED ISSUE #10204: MAINT: Remove tempita
For @hoechenberger, replace `mne/_tempita` (and refs to it) with some other templating system like `jinja2`. It's the one idea from #10198 that isn't covered by #10199.

### PR DESCRIPTION
**Pull request by hoechenberger**
Fixes #10204

WIP:
- [x] Render tutorial
- [x] Remove tempita from HTML templates for Jupyter
- [x] Changelog entry

Nice: VS Code has debugging support for Jinja templates:
Screen Shot 2022-01-16 at 14 12 22

### CODE REVIEW

**Comment by hoechenberger** [comment]
One circle job failed because MNE-BIDS depends on our censored Tempita. This will require an update of MNE-BIDS and can be ignored here for now. 

All other checks passed. 

I will update the changelog ASAP; in any case, this PR is ready for review. 

- Removed Tempita in favor of Jinja2
- HTML repr of Info is more robust 
- I could remove some manual string escaping from the Python code, as Jinja2 will do that work for us now
- All HTML templates are now located under `mne/html_templates`

File: mne/html_templates/report/forward.html.jinja [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        {% endfor %}
+      </div>
+    </button>
+  </div>
+
+  <div id="accordion-collapse-{{ id }}" class="accordion-collapse collapse show"
+    aria-labelledby="accordion-header-{{ id }}">
+    <div class="accordion-body">
+      {{repr | safe}}
+      {{sensitivity_maps |safe }}
```
Comment (hoechenberger) [inline mne/html_templates/report/forward.html.jinja:20]:
```suggestion
      {{sensitivity_maps | safe }}
```

**Comment by jasmainak** [comment]
This broke `mne-bids` :)

```py
In [1]: import mne_bids
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-b553cc368f26> in <module>
----> 1 import mne_bids

~/Documents/github_repos/mne-bids/mne_bids/__init__.py in <module>
      3 __version__ = '0.10.dev0'
      4 from mne_bids import commands
----> 5 from mne_bids.report import make_report
      6 from mne_bids.path import (BIDSPath, get_datatypes, get_entity_vals,
      7                            print_dir_tree, get_entities_from_fname,

~/Documents/github_repos/mne-bids/mne_bids/report.py in <module>
      9 
     10 import numpy as np
---> 11 from mne.externals.tempita import Template
     12 from mne.utils import warn, logger, verbose
     13 

ImportError: cannot import name 'Template' from 'mne.externals.tempita' (unknown location)
```

**Comment by hoechenberger** [comment]
> This broke `mne-bids` :)

Already fixed: https://github.com/mne-tools/mne-bids/pull/944

**Pull request merged**